qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -657,26 +668,29 @@ def on_current_changed(self, idx):
             log.webview.debug("on_current_changed got called with invalid "
                               "index {}".format(idx))
             return
-        if self._now_focused is not None and mode_on_change == 'restore':
-            current_mode = modeman.instance(self._win_id).mode
-            if current_mode not in modes_to_save:
-                current_mode = usertypes.KeyMode.normal
-            self._now_focused.data.input_mode = current_mode
 
         log.modes.debug("Current tab changed, focusing {!r}".format(tab))
         tab.setFocus()
 
+        prompt_modes = [usertypes.KeyMode.prompt, usertypes.KeyMode.yesno]
         modes_to_leave = [usertypes.KeyMode.hint, usertypes.KeyMode.caret]
-        if mode_on_change != 'persist':
-            modes_to_leave += modes_to_save
+
+        mm_instance = modeman.instance(self._win_id)
+        current_mode = mm_instance.mode
+        log.modes.debug("{}: mode before tab change: {}."
```

Comment:
That log message is weird to read for a human. What about "Mode before tab change: {} (mode_on_change = {})" or so?

Reply:
this is intended as a sentinel for tests waiting for log entries (since there is no way to tell when logs from a previous test will end-up in the next test's logs because of the asynchronous nature of qb)

Reply:
It's still also in the debug log and can be useful for debugging issues. Why not make it useful as a sentinel *and* for humans?

Reply:
good point!